### ISSUE TITLE
TCP listener creation failed when there is no TCP route defined

### ISSUE THREAD

**Issue by zhaohuabing** [envoyproxy/gateway#4680]
error:
[2024-11-08 04:34:43.214][1][warning][config] [source/extensions/config_subscription/grpc/grpc_subscription_impl.cc:138] gRPC config for type.googleapis.com/envoy.config.listener.v3.Listener rejected: Error adding/updating listener(s) gateway-conformance-infra/same-namespace/tls: error adding listener '0.0.0.0:10443': no filter chains specified

configuration:
```yaml
apiVersion: gateway.networking.k8s.io/v1
kind: Gateway
metadata:
  name: same-namespace
  namespace: gateway-conformance-infra
spec:
  gatewayClassName: "{GATEWAY_CLASS_NAME}"
  listeners:
    - name: tls
      protocol: TLS
      port: 443
      tls:
        mode: Passthrough
      allowedRoutes:
        namespaces:
          from: Same
```

**Comment by arkodg** [comment]
cc @juwon8891

**Comment by zhaohuabing** [comment]
@juwon8891 @arkodg this is not ipv6 related. A TCP listener without route will be translated to a xDS listener without filter chain and rejected by the Envoy.

It's a minor issue - there is no real impact as no route defined for a TCP listener.

**Comment by haq204** [comment]
I ran into this issue as well. Note that if a cloud provider NLB (AWS, GCP, etc.) is fronting Envoy Gateway it can cause health checks in the NLB to fail since there is no active listener

**Comment by arkodg** [comment]
hey good to see you here again @haq204, trying to get this resolved this week and added into v1.2.2
